### PR TITLE
Upgrading JMXFetch to 0.49.6

### PR DESCRIPTION
**Pull request by carlosroman**
### What does this PR do?

This PR updates JMXFetch to version `0.49.6`

### Motivation

This fixes a `NullPointerException` on JBoss when user and password not set

### Possible Drawbacks / Trade-offs

None.

### Describe how to test/QA your changes

Update the misbehaving-jmx-server docker-compose (l[ink](https://github.com/DataDog/jmxfetch/tree/master/tools/misbehaving-jmx-server)) to use the new RC or dev image and ensure that metrics are collected properly.

### CODE REVIEW

File: releasenotes/notes/jmxfetch-0.49.6-437d4a0a92d65725.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Upgraded JMXFetch to `0.49.5 <https://github.com/DataDog/jmxfetch/releases/0.49.6>`_ which fixes a `NullPointerException` on
```
Comment (vickenty) [inline releasenotes/notes/jmxfetch-0.49.6-437d4a0a92d65725.yaml:11]: RST uses double-backticks for verbatim text
```suggestion
    Upgraded JMXFetch to `0.49.5 <https://github.com/DataDog/jmxfetch/releases/0.49.6>`_ which fixes a ``NullPointerException`` on
```

**Review by vickenty — APPROVED** [review]

**Review by brett0000FF — APPROVED** [review]

**Review by alexn-dd — APPROVED** [review]

**Review by FlorentClarret — APPROVED** [review]

**Review by FlorentClarret — CHANGES_REQUESTED** [review]
Requesting a change since you already `/merge`

File: release.json [hunk 2]
```diff
@@ -47,8 +47,8 @@
         "INTEGRATIONS_CORE_VERSION": "7.56.0-rc.2",
         "OMNIBUS_SOFTWARE_VERSION": "7.56.0-rc.3",
         "OMNIBUS_RUBY_VERSION": "7.56.0-rc.1",
-        "JMXFETCH_VERSION": "0.49.5",
-        "JMXFETCH_HASH": "d29f945af0353905a5c801ab6b6ac027aa0cb27535470f464ca9254f6f36ef23",
+        "JMXFETCH_VERSION": "0.49.6",
```
Comment (chouetz) [inline release.json:50]: Shall we backport this to the `agent6` release branch?

**Review by chouetz — APPROVED** [review]
Ok from my side, and you need to merge @FlorentClarret's suggestion

File: releasenotes/notes/jmxfetch-0.49.6-437d4a0a92d65725.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+upgrade:
+  - |
+    Upgraded JMXFetch to `0.49.5 <https://github.com/DataDog/jmxfetch/releases/0.49.6>`_ which fixes a ``NullPointerException`` on
```
[the remark this answers is not in the document]
Comment (carlosroman) [reply releasenotes/notes/jmxfetch-0.49.6-437d4a0a92d65725.yaml:11]: Thank you for picking up on that 🤦🏽‍♂️

**Review by FlorentClarret — APPROVED** [review]

File: release.json (hunk 2 above)
Comment (carlosroman) [reply release.json:50]: It doesn't have any security updates so I don't think we need to. I'm not 100% sure the current guidelines on when/what should be back ported to Agent 6 or not.

**Pull request merged**